packages/taco/src/conditions/base/contract.ts

```diff
@@ -70,6 +70,31 @@ const functionAbiSchema = z
     },
   );
 
+function toJsonAbiFormat(humanReadableAbi: string): any {
+  const abiWithoutFunctionKeyword = humanReadableAbi.replace(
+    /^function\s+/,
+    '',
+  );
+  const fragment = ethers.utils.FunctionFragment.from(
+    abiWithoutFunctionKeyword,
+  );
+  const jsonAbi = JSON.parse(fragment.format(ethers.utils.FormatTypes.json));
+
+  delete jsonAbi.constant;
+  delete jsonAbi.payable;
+
+  jsonAbi.inputs = jsonAbi.inputs.map((input: any) => ({
+    ...input,
+    internalType: input.type,
+  }));
+  jsonAbi.outputs = jsonAbi.outputs.map((output: any) => ({
+    ...output,
+    internalType: output.type,
+  }));
```

Comment:
Why do we need to map `type` to `internalType`? To be consistent with JSON ABI schema?

Reply:
Yes!